testsuite/tests/typing-misc/records.ml

```diff
@@ -264,5 +265,6 @@ Line 1, characters 0-30:
 1 | type perm = d = {y:int; x:int}
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Error: This variant or record definition does not match that of type d
-       Fields number 1 have different names, x and y.
+       1. Fields have different names, x and y.
+       2. Fields have different names, y and x.
```

Comment:
Note: it may be possible to reconstruct arbitrary-moves after the fact, by fusing a deletion and an insertion together when their endpoint are compatible. This means that the edit distance computed is different (moves cost 2 instead of 1 as we could expect), but the user message is nice. I'm not sure if the distance difference is a big deal.

Reply:
There should be an example in the testsuite for a larger displacement, for example

```
{a; b; c; d; e}
{b; c; d; e; a}
```

Reply:
I consider that on this example there is a clear regression from a better error message to a worse error message. I expect that this kind of situation is easy for beginners to run into, so I would be in favor of fixing this before we merge. I think that there are easy fixes -- in particular, recognizing displacements on the final proposed edition (which corresponds to minimal edits when giving displacements a cost of 2) sounds rather easy to implement. (Trying to modify the edit-distance algorithm to provide optimal suggestions for cost-1 displacements is of unknown difficulty, so maybe that should not be a requirement.)

Reply:
The new version of this PR refines the error message into:

> Error: This variant or record definition does not match that of type d
       Fields x and y have been swapped.